### PR TITLE
Add Support for Reading KubeVirt Infra Namespace in Cloud Provider Spec 

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Adding support for passing the underlying KubeVirt infra namespace where KubeVirt resources are going to be created in the infra cluster.
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add `InfraNamespace` field in the KubeOne API to control machine-controller KubeVirt provider resources 
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

File: pkg/apis/kubeone/types.go [hunk 1]
```diff
@@ -439,7 +439,10 @@ type HetznerSpec struct {
 }
 
 // KubevirtSpec defines the Kubevirt provider
-type KubevirtSpec struct{}
+type KubevirtSpec struct {
+	// InfraNamespace
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go:443]: Can you please add a proper godoc-based description for this field? This description is used for the user-facing documentation.

File: addons/machinecontroller/deployment-controller.yaml [hunk 2]
```diff
@@ -54,6 +54,10 @@ spec:
             {{ end -}}
             - -join-cluster-timeout=15m
           env:
+{{ if .Config.CloudProvider.Kubevirt.InfraNamespace -}}
+            - name: POD_NAMESPACE
+              value: "{{ .Config.CloudProvider.Kubevirt.InfraNamespace }}"
+{{ end }}
```
Comment (kron4eg) [inline addons/machinecontroller/deployment-controller.yaml:60]:
```suggestion
{{ with .Config.CloudProvider.Kubevirt -}}
{{ with .InfraNamespace }}
            - name: POD_NAMESPACE
              value: "{{ . }}"
{{ end }}
{{ end }}
```

Comment (kron4eg) [reply addons/machinecontroller/deployment-controller.yaml:60]: That will fix the NPE

**Comment by xmudrii** [comment]
/release-note-edit
```release-note
Add `.cloudProvider.kubevirt.infraNamespace` field to the KubeOneCluster API used to control what namespace will be used by the KubeVirt provider to create and manage resources in the infra cluster, such as VirtualMachines and VirtualMachineInstances
```

**Pull request merged**